apps/hash-frontend/next.config.js

```diff
@@ -86,118 +66,133 @@ const pageEntityTypeId = `${frontendUrl}/@${systemUserShortname}/types/entity-ty
-          return [
-            {
-              /**
-               * allow fetching types as JSON from anywhere
-               * @see ./src/middleware.page.ts for middleware which serves the JSON
-               */
-              source: "/:shortname/types/:path*",
-              has: [
-                {
-                  type: "header",
-                  key: "accept",
-                  value: "(.*application/json.*)",
-                },
-              ],
-              headers: [
-                {
-                  key: "access-control-allow-origin",
-                  value: "*",
-                },
-              ],
-            },
-          ];
-        },
-        pageExtensions: [
-          "page.tsx",
-          "page.ts",
-          "page.jsx",
-          "page.jsx",
-          "api.ts",
-        ],
-
-        // We call linters in GitHub Actions for all pull requests. By not linting
-        // again during `next build`, we save CI minutes and unlock more feedback.
-        // Thus, we can get Playwright test results and Preview releases for WIP PRs.
-        eslint: { ignoreDuringBuilds: true },
-        typescript: { ignoreBuildErrors: true },
-
-        sentry: {
-          autoInstrumentServerFunctions: false,
-          hideSourceMaps: false,
-        },
+    /** @type {import('next').NextConfig} */
+    {
+      async rewrites() {
+        return [
+          {
+            source: "/pages",
+            destination: `/entities?entityTypeIdOrBaseUrl=${pageEntityTypeId}`,
+          },
+        ];
+      },
+      redirects() {
+        return [
+          {
+            source: "/settings/organizations/:shortname(^(?!new)$)",
+            destination: "/settings/organizations/:shortname/general",
+            permanent: true,
+          },
+        ];
+      },
+      async headers() {
+        return [
+          {
+            /**
+             * allow fetching types as JSON from anywhere
+             * @see ./src/middleware.page.ts for middleware which serves the JSON
+             */
+            source: "/:shortname/types/:path*",
+            has: [
+              {
+                type: "header",
+                key: "accept",
+                value: "(.*application/json.*)",
+              },
+            ],
+            headers: [
+              {
+                key: "access-control-allow-origin",
+                value: "*",
+              },
+            ],
+          },
+        ];
+      },
+      pageExtensions: ["page.tsx", "page.ts", "page.jsx", "page.jsx", "api.ts"],
+
+      // We call linters in GitHub Actions for all pull requests. By not linting
+      // again during `next build`, we save CI minutes and unlock more feedback.
+      // Thus, we can get Playwright test results and Preview releases for WIP PRs.
+      eslint: { ignoreDuringBuilds: true },
+      typescript: { ignoreBuildErrors: true },
 
-        experimental: {
-          allowMiddlewareResponseBody: true,
```

Comment:
Removed, returning a `NextResponse` is sufficient to allow this (I have tested that the 'return types as JSON' middleware still works)